core/modules/layout/plugins/renderers/layout_renderer_flexible.inc

```diff
@@ -75,7 +75,6 @@ class LayoutRendererFlexible {
     $this->admin = TRUE;
     $this->layout = &$layout;
     $this->layout_info = $template_info;
-    $this->flexible = TRUE;
```

Comment:
This seems useful, so why are we removing it completely? It seems to me that this is easier rather than having to test the `$edit_path` each time. Compare before:
```php
  if (!empty($variables['renderer']->flexible)) {
```

...to now:
```php
  $edit_path = "admin/structure/layouts/settings/flexible-template/$layout_name/configure";
  if (current_path() == $edit_path) {
```

I would very much rather we added something like a new `is_flexible` property that does this for people working on contrib/custom code.

Reply:
The problem is: we only need this info in the edit form.
It was sort of convenient to use a property on-the-fly, but that's not possible anymore with PHP 8.2.

Reply:
Fair enough. If we are absolutely sure that this won't be useful to custom/contrib, then I'm fine with it gone. Especially if there is already an easy/elegant way to tell whether the layout being rendered is flexible or "regular".

Reply:
The renderer already is "LayoutRendererFlexible" - I don't see any benefit in this property.
The info just triggers loading the editor in the form, and AFAIK no contrib will ever need to handle that.

Reply:
I agree that using a property to store this information that is used only for the edit form may be overkill. 

However, I disagree with the approach (in `layout.theme.inc` above) of determining whether this is a flexible layout\ by "rebuilding" the flexible template edit path and comparing it to the current path. One of the aspects of using classes is that they act a bit like black boxes where the outside world only "knows" whatever the class "wants it to know". So, in theory, the preprocess function has no way of knowing the href for the edit form - that's internal information to the class. "Guessing" the path and hard coding the guess is not good.

So, the outside world has 3 ways of knowing things about the  `LayoutRendererFlexible` class: its public properties, its public methods, and its class name. A public method (eg `isFlexible()`) would be great to create, but we would need to add the method to other classes like `LayoutRendererEditor()` to be consistent (I wished the original developers had based ALL renderer classes on a parent renderer base class that had all these common methods!!!).

So, the after discarding a public property and a public method, the third best option is to test the class name. I suggest using `is_a()`, which will work for `LayoutRendererFlexible` AND all future children that extend it. This would replace all the href building etc shown above in `layout.theme.inc`.

Reply:
And to respond to my own objection 😄  I guess one may say that all paths starting with `admin/structure/layouts/settings/flexible-template` are public information since they are defined in `layout_menu()`.  Still, I think testing for the class is cleaner than testing for the path.

Reply:
@argiepiano I haven't had a lot of time to look into it (and don't want to derail this topic) but have been thinking in the back of my mind about https://github.com/backdrop/backdrop-issues/issues/6039. One approach I was thinking of was creating a new base layout class that could include a property such as `->type` which in this case could be `flexible` and in the case of core layouts would simply be `layout` (and in other implementations could be `paragraphs` or `field` or ??). Of course, the current layout class couldn't be renamed but perhaps could retroactively extend the new `BaseLayout` (or whatever) class.

Reply:
@argiepiano I'd admit any time, that my fix isn't very elegant. :see_no_evil:  However, is_a() won't do the trick.

Looking where this dynamic property is used, should explain why: https://github.com/backdrop/backdrop/blob/1.x/core/modules/layout/layout.theme.inc#L329

As you can see, both conditions handle flexis, so what other criterion could help us here?

Reply:
Hm, no luck. Tried to compare $variables on admin/structure/layouts/settings/flexible-template/THING/configure (configure the flexi layout) and admin/structure/layouts/manage/THING (manage blocks on it) – and they're the same on both pages. :shrug: The path is the only obvious difference.

Reply:
We could re-use the $region_buttons variable, I guess. It's set in function renderRowButtons() and it's null on the "Manage blocks" page.

Not sure, if that's more elegant, but at least some lines below in layout.theme.inc it's also in use.

```
    if (isset($variables['renderer']->region_buttons)) {
      $variables['flexible_editor'] = TRUE;
      $flexible_layout = layout_flexible_tempstore_load($layout_name);
    }
    else {
      $variables['flexible_editor'] = FALSE;
      $flexible_layout = layout_flexible_template_load($layout_name);
    }
```
@argiepiano or @laryn what do you think? Better or worse?

Reply:
Hi @indigoxela. I was thinking of the following:

```php
 if (!empty(is_a($variables['renderer'], 'LayoutRendererFlexible')) {
      $variables['flexible_editor'] = TRUE;
      $flexible_layout = layout_flexible_tempstore_load($variables['layout_info']['name']);
    }
    else {
      $variables['flexible_editor'] = FALSE;
      $flexible_layout = layout_flexible_template_load($variables['layout_info']['name']);
    }
```

But you are saying that won't work? I haven't tried this solution, but I'm not sure I understand why that wouldn't work. Let me know what I'm missing...

Reply:
I've been doing some quick testing (without patching Backdrop, only changing the if block I pasted above), and it seems to work as expected. `LayoutRendererFlexible` is only used for the "flexible template editor" according to the help block for the class. This means that, when you are viewing a flexible template, Backdrop will use `LayoutRendererStandard`. The `if` block above catches this - so, when you are in the flexible layout template edit form, the if will catch the  `LayoutRendererFlexible` class and then load the template from "tempstore". When viewing a flexible template then this will go to the `else` statement and load the layout from the saved template. 

Unless I'm missing something obvious, I believe this is a better solution than checking the path. 

@laryn - regarding your suggestion, in this case, that won't work so well. The issue here is that Backdrop is using two different renderer classes depending on the situation. In both cases, the layout is using a flexible template - the `LayoutRendererFlexible` is used to render the template edit form, while `LayoutRendererStandard` is being used for the same flexible template, but only when viewing the page that uses it. So, the in principle, the `type` property for Layout base classes is great, but it won't solve the problem here.

Reply:
Now I get it... :tada:

> LayoutRendererFlexible is only used for the "flexible template editor"

That's right! Instead of checking for a specific property, it's cleaner to check for the class. On the "Manage blocks" form it's LayoutRendererEditor, only on the template config form it's LayoutRendererFlexible. (On the regular layout page it's LayoutRendererStandard.)

@argiepiano many thanks for your patience. I was sort of lost in the Layout class names. :laughing: